File: Movieweb/src/models/Director.ts
import type { Movie } from "./Movie";

export class Director {
    id: number;
    name: string;
    nationality?: string;
    age?: number;
    active?: boolean;
    movies: Movie[];

    constructor() {
        this.id = 0;
        this.name = "";
        this.movies = [];
    }
}

File: MoviesAPI/MoviesAPI/Models/Director.cs
using System;
using System.Collections.Generic;

namespace MoviesAPI.Models;

public partial class Director
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Nationality { get; set; }

    public int? Age { get; set; }

    public bool? Active { get; set; }

    public virtual ICollection<Movie> Movies { get; set; } = new List<Movie>();
}
